akka-http/src/main/scala/akka/http/scaladsl/server/directives/JwtDirectives.scala

```diff
@@ -0,0 +1,98 @@
+package akka.http.scaladsl.server.directives
+
+import akka.http.impl.util.enhanceString_
+import akka.http.scaladsl.common.{ NameDefaultReceptacle, NameOptionReceptacle, NameReceptacle, RequiredValueReceptacle }
+import akka.http.scaladsl.model.{ AttributeKey, HttpRequest }
+import akka.http.scaladsl.server.{ Directive, Directive0, Directive1, InvalidRequiredValueForQueryParamRejection, MalformedQueryParamRejection, MalformedRequestContentRejection, MissingQueryParamRejection, RequestContext }
+import akka.http.scaladsl.server.Directives.{ Authenticator, AuthenticatorPF, authenticateOAuth2, authenticateOAuth2Async, authenticateOAuth2PF, headerValueByName }
+import akka.http.scaladsl.server.directives.BasicDirectives.{ extractRequestContext, pass, provide }
+import akka.http.scaladsl.server.directives.FutureDirectives.onComplete
+import akka.http.scaladsl.server.directives.RouteDirectives.reject
+import akka.http.scaladsl.server.util.{ JwtSprayJson, JwtSupport }
+import akka.http.scaladsl.unmarshalling.Unmarshaller
+import akka.http.scaladsl.util.FastFuture
+import com.typesafe.config.Config
+import pdi.jwt.JwtClaim
+import spray.json.{ JsBoolean, JsNumber, JsObject, JsString, JsValue }
+
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.{ Await, Future }
+import scala.reflect.ClassTag
+import scala.util.{ Failure, Success }
+
+trait JwtDirectives {
+
+  import BasicDirectives._
+  import RouteDirectives._
+
+  def jwt(): Directive1[JwtClaims] = {
+    extractActorSystem.flatMap { system =>
+      jwt(system.settings.config)
+    }
+  }
+
+  private def jwt(settings: Config): Directive1[JwtClaims] = {
+    // FIXME: Should we use this or just manually export the header value? Is there a clever way to extract a realm that makes sense? path?
+    authenticateOAuth2("realm", bearerTokenAuthenticator(settings)).flatMap { claims =>
+      extractRequestContext.flatMap { ctx =>
+        provide(new JwtClaims(ctx, claims))
+      }
+    }
+  }
```

Comment:
I think these two should be `jwt(): Directive1[JwtClaims]` and a public `jw(JwtSettings): Directive1[JwtClaims]` for users that want to programatically configure